src/api/identity/group/requests.rs

```diff
@@ -0,0 +1,223 @@
+use super::responses::{
+    CreateGroupResponse, ListGroupsByIdResponse, ListGroupsByNameResponse, ReadGroupByIdResponse,
+    ReadGroupByNameResponse,
+};
+use rustify_derive::Endpoint;
+use serde::Serialize;
+use std::{collections::HashMap, fmt::Debug};
+
+/// ## Create an group
+///
+/// This endpoint creates or updates a group.
+///
+/// * Path: identity/group
+/// * Method: POST
+/// * Reference: https://developer.hashicorp.com/vault/api-docs/secret/identity/group#create-a-group
+#[derive(Builder, Debug, Default, Endpoint, Serialize)]
+#[endpoint(
+    path = "identity/group",
+    method = "POST",
+    builder = "true",
+    response = "CreateGroupResponse"
+)]
+#[builder(setter(into, strip_option), default)]
+pub struct CreateGroupRequest {
+    /// Name of the group. If set (and ID is not set), updates the corresponding existing group.
+    pub name: Option<String>,
+    /// ID of the group. If set, updates the corresponding existing group.
+    pub id: Option<String>,
+    /// Type of the group, internal or external. Defaults to internal.
+    #[serde(rename = "type")]
+    pub group_type: Option<String>,
+    /// Metadata to be associated with the entity.
+    pub metadata: Option<HashMap<String, String>>,
+    /// Policies to be tied to the group.
+    pub policies: Option<Vec<String>>,
+    /// Group IDs to be assigned as group members.
+    pub member_group_ids: Option<Vec<String>>,
+    /// Entity IDs to be assigned as group members.
+    pub member_entity_ids: Option<Vec<String>>,
+}
+
+/// ## Read group by ID
```

Comment:
This makes the `CreateGroupRequest` convertible into an `UpdateGroupByIdRequest`
```suggestion
}

impl Into<UpdateGroupByIdRequest> for CreateGroupRequest {
    fn into(self) -> UpdateGroupByIdRequest {
        UpdateGroupByIdRequest{
            id: self.id.unwrap(),
            name: self.name,
            group_type: self.group_type,
            metadata: self.metadata,
            policies: self.policies,
            member_group_ids: self.member_group_ids,
            member_entity_ids: self.member_entity_ids,
        }
    }
}

/// ## Read group by ID
```